src/metabase/models/serialization/util.clj

```diff
@@ -328,6 +345,10 @@
          ["field"    (field :guard vector?) tail] (into #{(field->path field)} (mbql-deps-map tail))
          [:field-id  (field :guard vector?) tail] (into #{(field->path field)} (mbql-deps-map tail))
          ["field-id" (field :guard vector?) tail] (into #{(field->path field)} (mbql-deps-map tail))
+         [:metric    (field :guard portable-id?)] #{[{:model "Metric" :id field}]}
+         ["metric"   (field :guard portable-id?)] #{[{:model "Metric" :id field}]}
+         [:segment   (field :guard portable-id?)] #{[{:model "Segment" :id field}]}
+         ["segment"  (field :guard portable-id?)] #{[{:model "Segment" :id field}]}
```

Comment:
Sorry for not catching this on the last PR review. But this function looks busted. It's trying to match old-style MBQL clauses like `:field-id` as well as both string and keyword versions. Why not run your MBQL thru `metabase.mbql.normalize` first so you only have to worry about matching the current version of the MBQL clauses, and only have to match the keyword vectors?

Reply:
I'm going to push this to a later cleanup to avoid blocking this change.

I think it actually runs deeper than that - I think this code is left over from when we were doing raw selects, not Toucan-backed ones, and therefore got the raw JSON strings for `:dataset_query` and `:visualization_settings`. I think now those are serialized as nicely formed map structures instead of strings. But I don't want to block this PR on testing and investigating that.